### PR TITLE
Added prometheus pod check for collect prometheus pod

### PR DESCRIPTION
**Pull request by agopala95**
Added a check to ensure prometheus pods are up and running before executing extraction of prometheus data within the "collect-prometheus" pod

### CODE REVIEW

File: pkg/e2e/state/prometheus.go [hunk 1]
```diff
@@ -26,6 +31,27 @@ var _ = ginkgo.Describe(clusterStateInformingName, func() {
...
 		cmd := promCollectCmd + " > " + runner.DefaultRunner.OutputDir + "/prometheus.tar.gz; err=$? ; if (( $err != 0 )) ; then exit $err ; fi"
 
+		poderr := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+			podCount := 0
+			var podErr error
+			list, _ := verify.FilterPods("openshift-monitoring", "app=prometheus", h)
+			names, podNum := verify.GetPodNames("prometheus-operator", list, h)
+			if podNum > 0 {
+				for _, value := range names {
+					if (value == "prometheus-k8s-0") || (value == "prometheus-k8s-1") {
```
Comment (whereswaldon) [inline pkg/e2e/state/prometheus.go:41]:
```suggestion
					if strings.HasPrefix("prometheus-k8s-", value) {
```
Technically doesn't doesn't check the number at all, but I'm not sure that matters. You can take this or leave it.

File: pkg/e2e/state/prometheus.go [hunk 2]
```diff
@@ -26,6 +31,27 @@ var _ = ginkgo.Describe(clusterStateInformingName, func() {
...
+						podCount += 1
+					}
+				}
+				if podCount == 2 {
```
Comment (whereswaldon) [inline pkg/e2e/state/prometheus.go:45]:
```suggestion
				if podCount >= 2 {
```
What if the default number of pods increases in the future?

File: pkg/e2e/state/prometheus.go [hunk 3]
```diff
@@ -26,6 +31,27 @@ var _ = ginkgo.Describe(clusterStateInformingName, func() {
...
+					podErr = nil
+					return true, podErr
+				}
+			}
+			podErr = errors.New("Prometheus pods are not in running state")
+			return false, podErr
```
Comment (whereswaldon) [inline pkg/e2e/state/prometheus.go:51]:
```suggestion
			return false, errors.New("Prometheus pods are not in running state")
```

File: pkg/e2e/state/prometheus.go [hunk 4]
```diff
@@ -26,6 +31,27 @@ var _ = ginkgo.Describe(clusterStateInformingName, func() {
...
+			names, podNum := verify.GetPodNames("prometheus-operator", list, h)
+			if podNum > 0 {
+				for _, value := range names {
+					if (value == "prometheus-k8s-0") || (value == "prometheus-k8s-1") {
+						podCount += 1
+					}
+				}
+				if podCount == 2 {
+					podErr = nil
+					return true, podErr
```
Comment (whereswaldon) [inline pkg/e2e/state/prometheus.go:47]:
```suggestion
					return true, nil
```

File: pkg/e2e/state/prometheus.go [hunk 5]
```diff
@@ -26,6 +31,27 @@ var _ = ginkgo.Describe(clusterStateInformingName, func() {
 		// common for a running prometheus instance
 		cmd := promCollectCmd + " > " + runner.DefaultRunner.OutputDir + "/prometheus.tar.gz; err=$? ; if (( $err != 0 )) ; then exit $err ; fi"
 
+		poderr := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+			podCount := 0
+			var podErr error
+			list, _ := verify.FilterPods("openshift-monitoring", "app=prometheus", h)
```
Comment (whereswaldon) [inline pkg/e2e/state/prometheus.go:37]: Why discard the error here?

File: pkg/e2e/state/prometheus.go [hunk 6]
```diff
@@ -26,6 +31,27 @@ var _ = ginkgo.Describe(clusterStateInformingName, func() {
 		// common for a running prometheus instance
 		cmd := promCollectCmd + " > " + runner.DefaultRunner.OutputDir + "/prometheus.tar.gz; err=$? ; if (( $err != 0 )) ; then exit $err ; fi"
 
+		poderr := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+			podCount := 0
+			var podErr error
```
Comment (whereswaldon) [inline pkg/e2e/state/prometheus.go:36]: I don't think you need this variable, though perhaps I'm missing something.

**Review by whereswaldon — CHANGES_REQUESTED** [review]
I think there are some opportunities to simplify, but this looks solid.

File: pkg/e2e/state/prometheus.go (hunk 1 above)
Comment (agopala95) [reply pkg/e2e/state/prometheus.go]: Yeah this makes sense. I've updated the check.

File: pkg/e2e/state/prometheus.go (hunk 2 above)
Comment (agopala95) [reply pkg/e2e/state/prometheus.go]: That could be a possibility. I've implemented the change.

File: pkg/e2e/state/prometheus.go (hunk 6 above)
Comment (agopala95) [reply pkg/e2e/state/prometheus.go]: Yup I guess we wouldn't be needing this variable.

File: pkg/e2e/state/prometheus.go (hunk 3 above)
Comment (agopala95) [reply pkg/e2e/state/prometheus.go]: Changes done!

File: pkg/e2e/state/prometheus.go (hunk 5 above)
Comment (agopala95) [reply pkg/e2e/state/prometheus.go]: The FilterPods function prints an error so I thought that would be enough but I guess including the error here is far more robust.

**Review by whereswaldon — COMMENTED** [review]
/lgtm

**Pull request merged**